### PR TITLE
Prevent bonding issue by removing extra network interface from bridge_ports on sle12 autoyast

### PR DESCRIPTION
**Pull request by Julie-CAO**
The incorrect network bridge configuration is causing problem:
```
The machine bare-metal2.oqa.prg2.suse.org act as router and created illegal duplicate traffics for gateway of the oqa.prg2.suse.org subnet.
We sow "DUP" responses for ping toward vlan gateway 10.145.10.254 not just in k2 VM but also in other machines like for example on DHCP servers.

We investigated the traffics and found evidences about problematic traffics in J11 TOR switches specifically on port 5 on both TORs. ports are used by the server bare-metal2.oqa.prg2.suse.org as eth0 and eth1 see: https://racktables.nue.suse.com/index.php?page=object&object_id=23403
Our suspicion is incorrect setup bond between eth0 and eth1 interfaces that want to be LACP bond, but that is not setup on our side and in he end act as loop in this vlan.
```

```
bare-metal2:/etc/sysconfig/network # cat ifcfg-br0
BOOTPROTO='dhcp'
BRIDGE='yes'
BRIDGE_FORWARDDELAY='15'
BRIDGE_PORTS='eth0 eth1 em1 em2'
BRIDGE_STP='off'
STARTMODE='auto'
```

The culprit is `BRIDGE_PORTS='eth0 eth1 em1 em2'`.   "make a bridge from two interfaces is bad practice"

- Verification run: 
[sle12sp5_kvm_on_gonzo](https://openqa.suse.de/tests/14912427)
[sle12sp5_kvm_on_fozzie](https://openqa.suse.de/tests/14912432)
[sle12sp5_kvm_on_amd-zen3](http://openqa.suse.de/tests/14912498)
[sle12sp5_kvm_on_blackbauhinia](http://openqa.suse.de/tests/14912509)
[sle12sp5_xen_on_fibnacci](https://openqa.suse.de/tests/14912467)

### CODE REVIEW

File: data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep
Comment (alice-suse) [inline data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:113]: @Julie-CAO Thanks for the fix! Code LGTM. It would be even better if a comment can be added here to explain how it is supposed to be set for bridge ports.

File: data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep
Comment (waynechen55) [inline data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:114]: I am curious about this pattern ```/((?:[0-9a-fA-F]{2}:){5}[0-9a-fA-F]{2})/```. What are you trying to match >?

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:114]: the MAC of ipv4 or ipv6. 
```
~/gitcode/salt-pillars-openqa/openqa> grep SUT_NETDEVICE workerconf.sls 
        SUT_NETDEVICE: ec:2a:72:02:84:20
        SUT_NETDEVICE: eth1
        SUT_NETDEVICE: F4:02:70:B9:3A:66
```

File: data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep
Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:113]: done

File: data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep
Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: add inline comment as well.

Comment (waynechen55) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: Why not just give the real device name to ```SUT_NETDEVICE``` ? So you can put value of it in ```bridge_ports``` directly. I do not think it makes any sense to differentiate the device name based on whether ```SUT_NETDEVICE``` is given mac address or real device name. It should always has device name.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: because `SUT_NETDEVICE` is used elsewhere, fg. the bootscript to install the host, 
```
kernel http://openqa.qa2.suse.asia/assets/repo/fixed/SLE-12-SP5-Server-DVD-x86_64-GM-DVD1/boot/x86_64/loader/linux install=http://openqa.qa2.suse.asia/assets/repo/fixed/SLE-12-SP5-Server-DVD-x86_64-GM-DVD1?device=B8:CB:29:DB:C1:7A ifcfg=B8:CB:29:DB:C1:7A=dhcp4   root=/dev/ram0 initrd=initrd  textmode=1
```

Comment (waynechen55) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: Anyway there is no logic or proper relation between "mac being used" and "em1 interface". I do not think you should use it this way.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: It has been being used this way in test scripts. I don't know which tests or which teams are using it. My PR is the quickest way to fix the bonding issue to make our tests workable meanwhile not make other tests broken.

Comment (alice-suse) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: @Julie-CAO @waynechen55   To me, https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19726/files#r1677298216 is valid point. SUT_NETDEVICE may not always be em1, although maybe now it is the value. 
On the other hand, I understand that it is not easy to detect the real name of the SUT_NETDEVICE at prepare_profile stage if it is given mac address. 

To find an easy solution, how do you think about introducing one more variable to specify the bridge port for our dev_host_xx.xml.ep?

Comment (waynechen55) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: I think introduce a new setting ```VIRT_HOST_BRIDGE_PORT``` is better. It clearly specify its usage and avoid ambiguity.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: I agree that introducing a new setting for sure is the perfect solution, if you all think it is no hurry to get the issue fixed, and we have only sle15sp7 one release need this setting. There are what we need to do:

- submit a MR to add the new setting to workers.ini
- adapt our automation tests with these two settings. I guest a few scripts need to modify, ipxe_install.pm, boot_from_pxe.pm, 2 autoyast templates, one or two other feature tests scripts.
- an entire verification tests

Based on these changes above, if the time is ok, I'd like to know which task should I do firstly.

- this PR
- a bug follow up https://bugzilla.suse.com/show_bug.cgi?id=1221812#c19
- vgpu automation
- fix bare-metal 3 boot order

Comment (alice-suse) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: It seems more work than expected :(. Sorry for overlooking some details. @Julie-CAO  How long do you expect it to need?

I do not quite follow details about that infra ticket related with this PR. Is it correct that without the fix, we will impact other team's infra/automation? If yes, I think fixing this is first priority. It's fine to merge it now, then go directly with enhancement.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: > It seems more work than expected :(. Sorry for overlooking some details. @Julie-CAO How long do you expect it to need?

It will cost a few days(also considering OSD is not fuctioning well recently).

> I do not quite follow details about that infra ticket related with this PR. Is it correct that without the fix, we will impact other team's infra/automation? If yes, I think fixing this is first priority. It's fine to merge it now, then go directly with enhancement.

No, if we don't run any sle12sp5 autoyast installation on those 3 machines I listed in slack, we will not impact the subnet or other teams. The impacts to us: 

- we can't run sle12sp5 host installation tests. 
- bare-metal2 is disconnected.

Comment (waynechen55) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: You can keep ```SUT_NETDEVICE``` anywhere else. Just introducing a new setting for new usage. I do not think it is good to merge this without this change. I do not think there is any urgent requirement to merge this pull request at the moment.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: > You can keep `SUT_NETDEVICE` anywhere else. Just introducing a new setting for new usage.

No, this setting is used both as interface name and MAC in test scripts. We need split them completely since we will introduce a new setting.

Comment (waynechen55) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: > > You can keep `SUT_NETDEVICE` anywhere else. Just introducing a new setting for new usage.
> 
> No, this setting is used both as interface name and MAC in test scripts. We need split them completely since we will introduce a new setting.

I still do not think you should rush with compromised quality. If this is very urgent, the make this the highest priority task. And as I said, also good opportunity of refining relevant settings for machines.

Comment (alice-suse) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: > > You can keep `SUT_NETDEVICE` anywhere else. Just introducing a new setting for new usage.
> 
> No, this setting is used both as interface name and MAC in test scripts. We need split them completely since we will introduce a new setting.

@Julie-CAO 
Not sure if we are on the same page... I think, in the final solution, we can keep `SUT_NETDEVICE` untouched. So wherever using it, keeping it the way they are. For the newly introduced setting, use it for port purpose only in the autoyast. It is just that, the value assigners/users, shall make them matched. That is the SUT_NETDEVICE(name or mac) corresponds to the NIC name given in port.

Given the impact listed in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/19726#discussion_r1678697260, I think it is worthwhile to fix this first. I am fine with doing it with two steps if you want.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: so, do you mean keep `SUT_NETDEVICE` the current value(some are NIC names and some are ports) and add `BRIDGE_PORTS` to these 3 machines? I am ok with that.

```
~/gitcode/salt-pillars-openqa/openqa> grep SUT_NETDEVICE workerconf.sls 
        SUT_NETDEVICE: ec:2a:72:02:84:20
        SUT_NETDEVICE: eth1
        SUT_NETDEVICE: F4:02:70:B9:3A:66
```

will be changed to be:

```
~/gitcode/salt-pillars-openqa/openqa> grep -e SUT_NETDEVICE -e BRIDGE_PORTS workerconf.sls 
        SUT_NETDEVICE: ec:2a:72:02:84:20
        BRIDGE_PORTS: em1

        SUT_NETDEVICE: eth1
        BRIDGE_PORTS: eth1

        SUT_NETDEVICE: F4:02:70:B9:3A:66
        BRIDGE_PORTS: em1
```

Comment (waynechen55) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: I prefer "without s".

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: > I prefer "without s". This is very trivial.

Edited in my previous comments. 

After Alice confirm, I'll begin with the change, and annouce this new setting in team for some machines in BJ after the PR is merged.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: I am ok with any of your solutions but If it will take much time, I need reset the priority of my tasks in hand.

Comment (alice-suse) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: @Julie-CAO Yes.

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:116]: Ok

**Comment by Julie-CAO** [comment]
The verification runs with the new introduced setting `BRIDGE_PORT`: 
[sle12sp5_kvm_on_unreal3https://openqa.suse.de/tests/14940223)
[sle12sp5_kvm_on_gonzo](https://openqa.suse.de/tests/14940212)
[sle12sp5_kvm_on_amd-zen3](https://openqa.suse.de/tests/14940218)
[sle12sp5_kvm_on_blackbauhinia](https://openqa.suse.de/tests/14941807)
[sle12sp5_kvm_on_ph052](http://10.67.129.27/tests/436)

And please merge this PR after https://gitlab.suse.de/openqa/salt-pillars-openqa/-/merge_requests/867 is merged.

File: data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep
Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep]: `<%= $get_var->('BRIDGE_PORT', 'eth0') %>` 

Yes, I noticed it and was verifing,  http://10.67.129.27/tests/437, it worked. Thanks.

**Review by Martchus — APPROVED** [review]

File: data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep
Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep]: fixed, thanks.

**Review by alice-suse — APPROVED** [review]
lgtm, thanks for the quick fix

File: data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep
Comment (waynechen55) [inline data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:113]: If BRIDGE_PORT use it, otherwise use SUT_NETDEVICE ?

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:113]: No, the bridge is default to `eth0` if there is no `BRIDGE_PORT`.

`<bridge_ports><%= $get_var->('BRIDGE_PORT', 'eth0') %></bridge_ports>`

Comment (waynechen55) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:113]: Why not use SUT_NETDEVICE if BRIDGE_PORT is not present ?

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:113]: `SUT_NETDEVICE` is an optional setting, and its value might be a MAC instead of an interface name.

Comment (waynechen55) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:113]: Yes. When it is MAC, you use BRIDGE_PORT.  When BRIDGE_PORT is not present, use SUT_NETDEVICE instead. Right ?

Comment (Julie-CAO) [reply data/virt_autotest/host_unattended_installation_files/autoyast/dev_host_12.xml.ep:113]: ?? it is as complex as the way without the new setting. Since we have introduced a new setting,  we need not implement it so complicatedly, this logic only saves one `BRIDGE_PORT` on one machine `gonzo`. If you realy bother with two more settings, there is no reason that we introduced a new one.

**Pull request merged**
